gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -542,17 +560,32 @@ public void start(Listener<RespT> responseListener, Metadata headers) {
             new SimpleForwardingClientCallListener<RespT>(responseListener) {
               @Override
               public void onClose(Status status, Metadata trailers) {
-                try {
+                if (wasClosed.compareAndSet(false, true)) {
                   super.onClose(status, trailers);
-                } finally {
-                  entry.release();
+                  if (wasReleased.compareAndSet(false, true)) {
+                    entry.release();
```

Comment:
This needs to be in a finally block so that ref count can be decremented even if the caller's responseListener throws an error. In other words it doesnt matter if the caller's handler has a bug, from the channel perspective, the call is done and the ref count should be decremented

Reply:
Updated